route/src/route_generator_worker.cpp

```diff
@@ -438,5 +442,77 @@ namespace route {
         return true;
     }
 
+bool RouteGeneratorWorker::crossTrackErrorCheck(const geometry_msgs::PoseStampedConstPtr& msg, lanelet::ConstLanelet current)
+{
+  if (!world_model_->getMap()) 
+  {
+    ROS_ERROR_STREAM("WMListener received a route before a map was available. Dropping route message.");
+    return true;
+  }
+
+  auto route = world_model_->getRoute()->shortestPath();
+
+  bool out_of_llt_bounds = false;
+
+    lanelet::BasicPoint2d position;
+
+    position.x()= msg->pose.position.x;
+    position.y()= msg->pose.position.y;  
+
+    if (boost::geometry::distance(position, current.polygon2d()) > cross_track_dist)
+        {
+            cte_count++;
+
+            if(cte_count > 4)
+                return true;
+        }
+
+  if(boost::geometry::within(position, current.polygon2d()) == false) //Determine whether or not the vehicle is in the lanelet polygon
+      {
+        out_of_llt_bounds = true;
+      }
+
+    auto following_llts = world_model_->getMapRoutingGraph()->following(current);
+    lanelet::ConstLanelets filtered_llts;//TODO: Filter lanelets from following lanelets that are not a part of the route
+ 
+
+    for(auto i:following_llts)
+    {
+       for(auto j: route)
+       {
+           if (i.id() == j.id())
+            filtered_llts.push_back(i);
```

Comment:
i think we need to pass to this route object to be abel to use it. Anyways, on the second thought, if we are using route.get().shortestPath() I think we are guaranteed to have only 1 following lanalet. So we can just end the loop if we found the next one.